Source/Project64/UserInterface/MainWindow.cpp

```diff
@@ -973,7 +973,7 @@ LRESULT CALLBACK CMainGui::MainGui_Proc(HWND hWnd, UINT uMsg, WPARAM wParam, LPA
             case JSAPI_ACT_RESET:
                 if (g_BaseSystem)
                 {
-                    g_BaseSystem->ExternalEvent((bool)lParam ? SysEvent_ResetCPU_Soft : SysEvent_ResetCPU_Hard);
```

Comment:
I assume adding the (bool) was to fix a warning, probably better is "lParam != 0"

Reply:
Even that is overkill? LPARAM is LONG_PTR which is long or int64.

Reply:
I would think when lParam is 0 then SysEvent_ResetCPU_Soft , otherwise SysEvent_ResetCPU_Hard .. to do x ? y : z, x has to be a bool so it is not really overkill it just making sure the types are correct

Reply:
> to do x ? y : z, x has to be a bool

No? From any arbitrary C standard:

 > The first operand shall have scalar type.
 > The second operand is evaluated only if the first compares unequal to 0; the third operand is evaluated
only if the first compares equal to 0 ...

    int x = ...;
    return x ? a : b; // This is perfectly legal.

If you change it to `(bool)x` then it might even depend on your C++ language version? bool is more strict now than it used to be. The compiler might do `x & 1` instead of `x != 0`.